iceoryx_examples/automotive_soa/include/owl/kom/field_subscriber.inl

```diff
@@ -0,0 +1,173 @@
+               })
+               .or_else([](auto& result) {
+                   if (result != iox::popo::ChunkReceiveResult::NO_CHUNK_AVAILABLE)
+                   {
+                       std::cerr << "Error receiving chunk!" << std::endl;
+                   }
+               }))
+    {
+    }
+    return numberOfSamples;
+}
+
+template <typename T>
+inline Future<T> FieldSubscriber<T>::Get()
+{
+    // If we call Get() twice shortly after each other, once the response of the first request has not yet
+    // arrived, we have a problem
+    //! [FieldSubscriber get]
+    bool requestSuccessfullySend{false};
+    m_client.loan()
+        .and_then([&](auto& request) {
+            request.getRequestHeader().setSequenceId(m_sequenceId);
+            request.send().and_then([&]() { requestSuccessfullySend = true; }).or_else([](auto& error) {
+                std::cerr << "Could not send Request! Error: " << error << std::endl;
+            });
+        })
+        .or_else([&](auto& error) {
+            std::cerr << "Could not allocate Request! Error: " << error << std::endl;
+            requestSuccessfullySend = false;
+        });
+
+    if (!requestSuccessfullySend)
+    {
+        return Future<FieldType>();
+    }
+    return receiveResponse();
+    //! [FieldSubscriber get]
+}
+
+template <typename T>
+inline Future<T> FieldSubscriber<T>::Set(const FieldType& value)
+{
+    // If we call Set() twice shortly after each other, once the response of the first request has not yet
+    // arrived, we have a problem
+    bool requestSuccessfullySend{false};
+    m_client
+        .loan()
+        //! [FieldSubscriber set]
+        .and_then([&](auto& request) {
+            request.getRequestHeader().setSequenceId(m_sequenceId);
+            request->emplace(value);
+            //! [FieldSubscriber set]
+            request.send().and_then([&]() { requestSuccessfullySend = true; }).or_else([](auto& error) {
+                std::cerr << "Could not send Request! Error: " << error << std::endl;
+            });
+        })
+        .or_else([&](auto& error) {
+            std::cerr << "Could not allocate Request! Error: " << error << std::endl;
+            requestSuccessfullySend = false;
+        });
+
+    if (!requestSuccessfullySend)
+    {
+        return Future<FieldType>();
+    }
+    return receiveResponse();
+}
+
+template <typename T>
+inline Future<T> FieldSubscriber<T>::receiveResponse()
+{
+    Promise<FieldType> promise;
+    auto future = promise.get_future();
+    // Typically you would e.g. use a worker pool here, for simplicity we use a plain thread
+    std::thread(
+        [&](Promise<FieldType>&& promise) {
+            // Avoid race if MethodClient d'tor is called while this thread is still running
+            //! [FieldSubscriber receive response]
+            std::lock_guard<iox::posix::mutex> guard(m_mutex);
+
+            auto notificationVector = m_waitset.timedWait(iox::units::Duration::fromSeconds(5));
```

Comment:
We should we output something on say `cerr` if we run into a timeout while waiting for the response. The magic number 5 is probably ok for example code, but a timeout case should be handled differently then a received response. Currently we would do nothing as far as I can see and return an invalid future that would never have a value.

Reply:
I do not know if there is a specific notification for this, but I would think if we wake up and find no notification it must have been a timeout. I am fairly sure spurious wake-ups are covered, so they cannot cause the wait to return.

Reply:
Added the error output.